runelite-api/src/main/java/net/runelite/api/queries/EquipmentItemQuery.java

```diff
@@ -25,78 +25,22 @@
 package net.runelite.api.queries;
 
 import net.runelite.api.Client;
-import net.runelite.api.widgets.Widget;
-import net.runelite.api.widgets.WidgetInfo;
-import net.runelite.api.widgets.WidgetItem;
+import net.runelite.api.InventoryID;
+import net.runelite.api.Item;
+import net.runelite.api.ItemContainer;
+import net.runelite.api.Query;
 
-import java.awt.Rectangle;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.Objects;
-
-public class EquipmentItemQuery extends WidgetItemQuery
+public class EquipmentItemQuery extends Query<Item, EquipmentItemQuery>
```

Comment:
I would probably rather see InventoryItemQuery constructor take in an InventoryID, and then just reuse that.

Reply:
Good call